lib/improver/tests/test_nbhood_utilities.py

```diff
@@ -0,0 +1,290 @@
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Unit tests for the nbhood.Utilities plugin."""
+
+
+import unittest
+
+from cf_units import Unit
+
+from iris.cube import Cube
+from iris.exceptions import CoordinateNotFoundError
+from iris.tests import IrisTest
+import numpy as np
+
+from improver.nbhood import Utilities
+from improver.tests.helper_functions_ensemble_calibration import (
+    add_forecast_reference_time_and_forecast_period)
+from improver.tests.test_nbhood_neighbourhoodprocessing import (
+    set_up_cube, set_up_cube_lat_long)
+
+
+class Test_cumulate_array(IrisTest):
+
+    """Test for cumulating an array vertically and horizontally."""
+
+    def test_basic(self):
+        """
+        Test that the vertical and horizontal accumulation produces the
+        intended result.
+        """
+        data = np.array([[5., 10., 14., 19., 24.],
+                         [4., 8., 11., 15., 19.],
+                         [3., 6., 8., 11., 14.],
+                         [2., 4., 6., 8., 10.],
+                         [1., 2., 3., 4., 5.]])
+        cube = set_up_cube(
+            zero_point_indices=((0, 0, 2, 2),), num_time_points=1,
+            num_grid_points=5)
+        result = Utilities.cumulate_array(cube)
+        self.assertIsInstance(result, Cube)
+        self.assertArrayAlmostEqual(result.data, data)
+
+    def test_for_multiple_times(self):
+        """
+        Test that the vertical and horizontal accumulation produces the
+        intended result when the input cube has multiple times.
+        """
+        data = np.array([[[5., 10., 14., 19., 24.],
+                          [4., 8., 11., 15., 19.],
+                          [3., 6., 8., 11., 14.],
+                          [2., 4., 6., 8., 10.],
+                          [1., 2., 3., 4., 5.]],
+                         [[5., 10., 15., 19., 24.],
+                          [4., 8., 12., 15., 19.],
+                          [3., 6., 9., 11., 14.],
+                          [2., 4., 6., 7., 9.],
+                          [1., 2., 3., 4., 5.]],
+                         [[4., 9., 14., 19., 24.],
+                          [4., 8., 12., 16., 20.],
+                          [3., 6., 9., 12., 15.],
+                          [2., 4., 6., 8., 10.],
+                          [1., 2., 3., 4., 5.]]])
+
+        cube = set_up_cube(
+            zero_point_indices=((0, 0, 2, 2), (0, 1, 3, 3), (0, 2, 0, 0)),
+            num_time_points=3, num_grid_points=5)
+        result = Utilities.cumulate_array(cube)
+        self.assertIsInstance(result, Cube)
+        self.assertArrayAlmostEqual(result.data, data)
+
+    def test_for_multiple_realizations_and_times(self):
+        """
```

Comment:
You don't need to test this as multiple realizations should not be allowed as the code currently stands. The test will need to be reintroduced with #75 

If you do leave this test in then the description is currently wrong